srtcore/sync.cpp

```diff
@@ -527,3 +536,120 @@ void srt::sync::CTimer::tick()
     m_event.notify_one();
 }
 
+
+void srt::sync::CGlobEvent::triggerEvent()
+{
+    return g_Sync.notify_one();
+}
+
+bool srt::sync::CGlobEvent::waitForEvent()
+{
+    return g_Sync.lock_wait_for(milliseconds_from(10));
+}
+
+////////////////////////////////////////////////////////////////////////////////
+//
+// CThread class
+//
+////////////////////////////////////////////////////////////////////////////////
+#ifndef USE_STDCXX_CHRONO
+
+srt::sync::CThread::CThread()
+{
+    m_thread = pthread_t();
+}
+
+srt::sync::CThread::CThread(void *(*start_routine) (void *), void *arg)
+{
+    create(start_routine, arg);
+}
+
+#if HAVE_FULL_CXX11
+srt::sync::CThread& srt::sync::CThread::operator=(CThread&& other)
+#else
+srt::sync::CThread& srt::sync::CThread::operator=(CThread& other)
+#endif
+{
+    if (joinable())
+    {
+        // If the thread has already terminated, then
+        // pthread_join() returns immediately.
+        // But we have to check it has terminated before replacing it.
+        LOGC(mglog.Error, log << "Assigning to a thread that is not terminated!");
```

Comment:
I think for situation like this you should add IPE keyword to the message.

Reply:
Added.